script.js

```diff
@@ -1,42 +1,74 @@
 "use strict";
 
-import { showAllData, resultFetchBreed } from "./fetchData.js";
+import { fetchBreedData, showAllData, clearData } from "./data.js";
 import { toggleImage } from "./toggleImage.js";
-import { searchBreed } from "./search.js";
-import { clearData } from "./clearData.js";
+import { inputValidation } from "./search.js";
 
 console.log("script");
 //*SELECTING ELEMENTS
 export const resultDataUl = document.querySelector("ul");
 export const searchInput = document.querySelector(".search__input");
-
+export const notification = document.querySelector(".notification");
 const searchButton = document.querySelector(".search__button");
 const clearButton = document.querySelector(".search__button-clear-all");
+const newLi = document.querySelector("li");
 
 //*FOCUS ON SEARCH INPUT
 export const inputFocus = function (searchInput) {
   searchInput.focus();
 };
 
 inputFocus(searchInput);
-//*ON WINDOW LOAD DISPLAY ALL THE DATA FROM THE showAllData FUNCTION
 
-window.addEventListener("load", showAllData(resultFetchBreed, resultDataUl));
+export const resultFetchBreed = await fetchBreedData(
+  resultDataUl,
+  notification,
+  searchInput
+);
 
-// const allLi = document.querySelectorAll(".result");
-// console.log(allLi);
+window.addEventListener(
+  "load",
+  fetchBreedData(resultDataUl, notification, searchInput)
+);
+
+//*ON WINDOW LOAD DISPLAY ALL THE DATA FROM THE showAllData FUNCTION
+window.addEventListener(
+  "load",
+  showAllData(resultFetchBreed, resultDataUl, newLi)
+);
 
 //*ON ul (closest li) CLICK EVENT TOGGLE IMAGES
-resultDataUl.addEventListener("click", toggleImage);
+const allLi = document.querySelectorAll(".result");
 
-searchButton.addEventListener("click", searchBreed);
+resultDataUl.addEventListener("click", function (e) {
+  toggleImage(e, allLi);
```

Comment:
> Še ena težava je, ko npr iščem določeno pasmo in mi izpiše zadetke, če togglam med slikami pasem, če kliknem neko pasmo mi prejšnje slike pasme ne zapre. Mogoče veš v čem je težava?

Predvidevam da je problem to, da vedno pošiljaš v `toggleImage` fn prvotne elemente k si jih selectal s tem: `const allLi = document.querySelectorAll(".result");` ampak to si naredu samo enkrat, so se je script.js file zagnal.. ne pa na vsak click, kjer kličeš `toggleImage` fn.

Drug problem bi bil implementacija `toggleImage` fn in sicer `e.target.closest('li').lastElementChild.classList.contains('show-image')` mi je mal sumljiv...
Kaj točno hočeš tukej narest, če mi lahko razložiš prosim.